### PR TITLE
Fix comment-tree pagination under-filling exact-fit subtrees

### PR DESCRIPTION
**Pull request by paskal**
### Problem

`Tree.limit()` paginates top-level comment subtrees against a comment budget. The overflow check used `>=`:

```go
if commentsCount+repliesCount >= limit && len(limitedNodes) > 0 { ... skip ... }
```

So a subtree that fit the page **exactly** was treated as overflow and dropped, under-filling the final page. Example: `limit=5` over two subtrees of sizes 3 and 2 returned only the first subtree (3 comments) and reported the second as `count_left`, even though 3+2 = 5 fits the requested limit precisely.

### Fix

Change the boundary to `>` so an exact-fit subtree is included:

```go
// a subtree that fits exactly is still included, and the first node is always returned in full.
if commentsCount+repliesCount > limit && len(limitedNodes) > 0 { ... skip ... }
```

The first node is still always returned in full even if it alone exceeds the limit, and larger subtrees still overflow to the next page. `count_left` / `last_comment` accounting is updated accordingly.

### Tests

- Adds table-driven tests for `MakeTree`'s limit/offset pagination and `countReplies`, including the exact-fit boundary cases.
- Updates the `/find` consistent-count golden expectations (`count_left`, `last_comment`) for the corrected boundary.

I wrote the original `>=` boundary, so this is my fix to it.

### CODE REVIEW

File: backend/app/store/service/tree_test.go [hunk 1]
```diff
@@ -152,6 +152,94 @@ func TestTreeSortNodes(t *testing.T) {
...
+		{Locator: loc, ID: "c4a", ParentID: "c4", Timestamp: ts(41)},
+		{Locator: loc, ID: "c4a1", ParentID: "c4a", Timestamp: ts(42)},
+	}
+	res := MakeTree(comments, "+time", 0, "")
+
+	byID := map[string]*Node{}
+	for _, n := range res.Nodes {
+		byID[n.Comment.ID] = n
+	}
+
```
[the remark this answers is not in the document]
Comment (paskal) [reply backend/app/store/service/tree_test.go:235]: Good call — added `require.Contains` for c1/c4 and `require.Len` on their replies before the `countReplies` calls and `Replies[0]` indexing, so a regression in `MakeTree` now fails with a clear assertion instead of a nil-pointer panic. Pushed in dbba809.

**Review by umputun — APPROVED** [review]
lgtm. Boundary flip is right, and the updated /find `count_left` / `last_comment` goldens derive correctly from the fixture, so this isn't a wrong fix hiding behind a re-baselined golden.
